mungegithub/mungers/e2e/e2e.go

```diff
@@ -231,21 +232,37 @@ func getJUnitFailures(r io.Reader) (failures map[string]string, err error) {
 		FailCount int        `xml:"failures,attr"`
 		Testcases []Testcase `xml:"testcase"`
 	}
-	ts := &Testsuite{}
-	// TODO: this full parse is a bit slower than the old scanf routine--
-	// could switch back for the case where we only care whether there was
-	// a failure or not if that is an issue in practice.
-	err = xml.NewDecoder(r).Decode(ts)
+	type Testsuites struct {
+		TestSuites []Testsuite `xml:"testsuite"`
+	}
+	var testSuiteList []Testsuite
+	failures = map[string]string{}
+	testSuites := &Testsuites{}
+	testSuite := &Testsuite{}
+	b, err := ioutil.ReadAll(r)
 	if err != nil {
 		return failures, err
 	}
-	if ts.FailCount == 0 {
-		return nil, nil
+	// first try to parse the result with <testsuites> as top tag
+	err = xml.Unmarshal(b, testSuites)
+	if err == nil && len(testSuites.TestSuites) > 0 {
+		testSuiteList = testSuites.TestSuites
+	} else {
+		// second try to parse the result with <testsuite> as top tag
+		xml.Unmarshal(b, testSuite)
+		if err != nil {
+			return nil, err
+		}
+		testSuiteList = []Testsuite{*testSuite}
 	}
-	failures = map[string]string{}
-	for _, tc := range ts.Testcases {
-		if tc.Failure != "" {
-			failures[fmt.Sprintf("%v {%v}", tc.Name, tc.ClassName)] = tc.Failure
+	for _, ts := range testSuiteList {
+		if ts.FailCount == 0 {
+			continue
+		}
+		for _, tc := range ts.Testcases {
+			if tc.Failure != "" {
+				failures[fmt.Sprintf("%v {%v}", tc.Name, tc.ClassName)] = tc.Failure
```

Comment:
There isn't anything in the XML itself to differentiate the runs right now (though we should probably look into adding that). Compare https://console.cloud.google.com/m/cloudstorage/b/kubernetes-jenkins/o/logs/kubernetes-test-go/12970/artifacts/junit_v1%2Cautoscaling-v1%2Cbatch-v1%2Cbatch-v2alpha1%2Cextensions-v1beta1%2Capps-v1alpha1%2Cfederation-v1alpha1%2Cpolicy-v1alpha1%2Crbac.authorization.k8s.io-v1alpha1_20160612-085102.xml and
https://console.cloud.google.com/m/cloudstorage/b/kubernetes-jenkins/o/logs/kubernetes-test-go/12970/artifacts/junit_v1%2Cextensions-v1beta1%2Cfederation-v1alpha1_20160612-084419.xml, which are part of the same Jenkins run.